### ISSUE TITLE
`InjectionPrincipalFactor` can error for D-classes of semigroups of matrices over finite fields

### ISSUE THREAD

**Issue by wilfwilson** [semigroups/Semigroups#764]
I found this by running `SemigroupsTestExtreme`:
```
gap> S := Semigroup(
> [Matrix(GF(3),
>    [[Z(3), Z(3), Z(3) ^ 0],
>      [0 * Z(3), Z(3), Z(3)],
>      [Z(3), 0 * Z(3), Z(3) ^ 0]]),
>   Matrix(GF(3),
>    [[Z(3), Z(3), 0 * Z(3)],
>      [Z(3) ^ 0, Z(3) ^ 0, 0 * Z(3)],
>      [Z(3) ^ 0, Z(3) ^ 0, 0 * Z(3)]])]);;
gap> InjectionPrincipalFactor(DClass(S, S.1));
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `^' on 2 arguments at /Users/Wilf/gap/lib/methsel2.g:249 called from
(mat[i][j] ^ -1) ^ inv at /Users/Wilf/gap/pkg/semigroups/gap/attributes/attr.gi:61 called from
SEMIGROUPS.InjectionPrincipalFactor( D, ReesMatrixSemigroup ) at /Users/Wilf/gap/pkg/semigroups/gap/attributes/attr.gi:651 called from
<function "InjectionPrincipalFactor for a Green's D-class (Semigroups)">( <arguments> )
```
My immediate guess is that it's something along the lines of:
* "function assumes that some group is a permutation group, which always used to be the case, but isn't necessarily, these days" or
* weirdness to do with indexing and submatrices.

But haven't looked into it.

**Comment by james-d-mitchell** [comment]
I didn't do anything explicitly to resolve this issue, but it doesn't exist any longer in #802 :)

**Comment by james-d-mitchell** [comment]
Resolved in v4.0.0.
